cpp/include/raft/spatial/knn/detail/ann_kmeans_balanced.cuh

```diff
@@ -421,32 +422,52 @@ auto adjust_centers(float* centers,
   return adjusted;
 }
 
-/** predict & adjust_centers combined in an iterative process. */
+/**
+ * @brief Expectation-maximization-balancing combined in an iterative process.
+ *
+ * Note, the `cluster_centers` is assumed to be already initialized here.
+ * Thus, this function can be used for fine-tuning existing clusters;
+ * to train from scratch, use `build_clusters` function below.
+ *
+ * @tparam T element type
+ *
+ * @param handle
+ * @param n_iters the requested number of iteration
+ * @param dim the dimensionality of the dataset
+ * @param[in] dataset a pointer to a managed row-major array [n_rows, dim]
+ * @param n_rows the number of rows in the dataset
+ * @param n_cluster the requested number of clusters
+ * @param[inout] cluster_centers a pointer to a managed row-major array [n_clusters, dim]
+ * @param[out] cluster_labels a pointer to a managed row-major array [n_rows]
+ * @param[out] cluster_labels a pointer to a managed row-major array [n_clusters]
+ * @param metric the distance type (there is a tweak in place for the similarity-based metrics)
+ * @param balancing_pullback
+ *   if the cluster centers are rebalanced on this number of iterations,
+ *   one extra iteration is performed (this could happen several times) (default should be `2`).
```

Comment:
Could you reword it to make it clearer? This is what I understand:
- without any rebalancing (adjustment), we make n_iters iteration,
- each adjustment will add extra iterations,
- The number of the additional iterations is `ballancing_pullback-1`

Reply:
Not exactly. The first and then every `ballancing_pullback`  rebalancing operations add one more iteration to the main cycle. I'll try to make it more understandable!